tokio/src/task/spawn.rs

```diff
@@ -127,7 +127,13 @@ cfg_rt! {
         T: Future + Send + 'static,
         T::Output: Send + 'static,
     {
-        spawn_inner(future, None)
+        // preventing stack overflows on debug mode, by quickly sending the
+        // task to the heap.
+        if cfg!(debug_assertions) {
```

Comment:
I think it would be reasonable to do something like this.
```suggestion
        if cfg!(debug_assertions) && std::mem::size_of::<T>() > 2048 {
```